### ISSUE TITLE
Spurious unused-import warning when imported variable is used as metaclass or as function annotation

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#143]
Originally reported by: **Antony Lee (BitBucket: [anntzer](http://bitbucket.org/anntzer), GitHub: @anntzer?)**

---

```
#!python
from abc import ABCMeta

class C(metaclass=ABCMeta):
    pass

def f(arg: ABCMeta):
    pass
```

Pylint doesn't detect either use of ABCMeta and warns that the import is unused.

I first created this issue on the astroid tracker (https://bitbucket.org/logilab/astroid/issue/14/spurious-unused-import-warning-when) but @PCManticore suggested that I move it here.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/143
